### PR TITLE
close for sockets

### PR DESCRIPTION
**Pull request by hiteshjoshi**

### CODE REVIEW

File: socket.go [hunk 1]
```diff
@@ -49,6 +51,11 @@ func newSocket(conn engineio.Conn, base *baseHandler) *socket {
 	return ret
 }
 
+func (s *socket) Close() error {
+	s.conn.Close()
```
Comment (mcastilho) [inline socket.go]: You are not checking there return of engine.Close() which returns an error

**Comment by googollee** [comment]
Why you want to add this Close method?

**Comment by greenenergy** [comment]
You need a close method so the server can close client connections. Otherwise, even when the client says "disconnect" the connection remains open. This uses server resources needlessly.

File: socket.go (hunk 1 above)
Comment (tehmaze) [inline socket.go]: I think you mean:

```go
func (s *socket) Close() error {
	return s.conn.Close()
}
```

**Pull request closed without merging**